node/src/components/small_network.rs

```diff
@@ -582,7 +582,7 @@ where
                 local_address=?transport.get_ref().local_addr(),
                 "connected outgoing to ourself - closing connection",
             );
-            return Effects::new();
+            return self.reconnect_if_not_connected_to_any_known_addresses(effect_builder);
```

Comment:
Why does the node even connect to itself in the first place? Is that b/c it may be in the `known_addresses`? Wouldn't it be simpler if the node filtered out its own address from that list?

Reply:
Not sure if it'd be simpler or not, but this is how the small_network has worked since inception I believe.  Perhaps that behaviour is due to change with the upcoming overhaul by @marc-casperlabs?

Either way, I'd be OK with changing the logic - just not in this PR :)